reference-delegate-explainer.md

```diff
@@ -0,0 +1,534 @@
+* When multiple elements are used for the same attribute, the author cannot control the order (the order is always the DOM order).
+
+### Use exported IDs instead of per-attribute targeting
+
+The [ExportID explainer](https://github.com/WICG/aom/blob/gh-pages/exportid-explainer.md) proposes a way to refer to elements inside the shadow DOM by name. For example, `"fancy-input::id(real-input)"` to refer to a specific `<input>` element inside a `<fancy-input>`.
+
+It would be possible to use exported IDs instead of `referenceDelegateMap` if/when it is necessary to refer to an element other than the primary reference delegate.
+
+#### Pros
+
+* Does not suffer from the [bottleneck effect](https://alice.pages.igalia.com/blog/how-shadow-dom-and-accessibility-are-in-conflict/#limitations-of-these-apis).
+* Potentially less confusing why you reference the _container_ listbox with `aria-activedescendant` instead of the element itself.
+
+#### Cons
+
+* Exposes some of the internal details of a control and does not give a way for the control to encapsulate those details.
+* Incompatible with ARIAMixin attributes, which don't allow directly referencing elements inside other shadow trees.
+   * It may be possible to work around this limitation, but it would require a change to the behavior of the ARIAMixin attributes, as well as new JavaScript APIs to resolve an IDREF like `"fancy-input::id(real-input)"` into an "ElementHandle" type object that references the element without giving full access to it (which would break shadow DOM encapsulation).
+
+## Appendix A: Combobox Example
+
+This "kitchen sink" example implements a `<fancy-combobox>` using two subcomponents: `<combobox-input>` and `<combobox-listbox>`. It demonstrates:
+* Delegating references through multiple layers of shadow DOM.
+   * A label in the light DOM refers to the `<input>` inside the `<combobox-input>`, which is itself inside the `<fancy-combobox>`.
+* Referring to an element in a sibling shadow tree. 
+   * Uses `ariaActiveDescendantElement` in `<combobox-input>` along with `referenceDelegateMap` in `<combobox-listbox>` to connect the `<input>` with a `<div role="option">`.
+* Using a custom prop to control the target of `referenceDelegateMap`.
+   * `<combobox-listbox>` allows the target of its `aria-activedescendant` to be controlled externally via its custom `activeitem` attribute.
+
+#### `<combobox-input>`
+
+This component is a wrapper around an `<input>` for use by a combobox.
+
+1. It delegates all references to the input. This lets, for example, a label for this component to be applied to the input.
+2. A custom attribute `listbox` is hooked up to both `ariaControlsElements` and `ariaActiveDescendantElement`.
+   * The listbox delegates the two attributes to different elements inside (see `<combobox-listbox>` below), but this component references the parent listbox for both.
+
+```html
+<template id="t-combobox-input">
+  <input id="real-input" role="combobox" type="text" />
```

Comment:
This requires that you have one input for your comboboxes and a different input for other things. Is there no path therein this could be a `<fancy-input>`?

Reply:
I think I addressed this in my reply here https://github.com/WICG/aom/pull/207#discussion_r1425977603, but let me know if you were asking about something else.

Reply:
I could update the combobox example to show how you could set the `role` of the input using a custom observed property on `fancy-input` (rather than a "new" `combobox-input` control). It's not new or specific to this proposal, but perhaps it helps show a more complete picture of referenceDelegate fitting in with other features.

Reply:
I've updated the combobox example to use a `<fancy-input>` that's not combobox-specific, instead of a specialized `<combobox-input>`. It's not fully-featured since it's just demoware, but hopefully it demonstrates the direction you could go to use a reusable custom input control for a combobox.